lab-beeksma/test/server.test.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const app = require('../server');
+const request = require('supertest')(app);
+const { expect } = require('chai');
+
+
+
+describe('Routes', function (){
+  it('should return routed for /', function (done){
+    request.get('/')
+      .expect(200)
+      .expect('routed')
+      .expect('content-type', 'text/plain')
+      .expect(res => {
+        expect(res.files).to.be.undefined;
+      })
+      .end(done);
+  });
+
+  it('should return Not Found for GET missing path', function (done){
+    request.get('/404')
+      .expect(404)
+      .expect('Not Found')
+      .expect('content-type', 'text/plain')
+      .end(done);
+  });
+
+  it('should return Not Found for POST missing path', function (done){
+    request.post('/404')
+      .expect(404)
+      .expect('Not Found')
+      .expect('content-type', 'text/plain')
+      .end(done);
+  });
+
+  it('should return Not Found for PUT missing path', function (done){
+    request.put('/404')
+      .expect(404)
+      .expect('Not Found')
+      .expect('content-type', 'text/plain')
+      .end(done);
+  });
+
+  it('should return Not Found for DELETE missing path', function (done){
+    request.delete('/404')
+      .expect(404)
+      .expect('Not Found')
+      .expect('content-type', 'text/plain')
+      .end(done);
+  });
+});
+
+describe('Simple Resource', function (){
+  var note = null;
+
+  describe('POST /note', function() {
+    it('should return bad request if no body', function (done){
+      request.post('/note')
+        .expect(400)
+        .expect('Bad Request : Error: expected item')
+        .expect('content-type', 'text/plain')
+        .end(done);
+    });
+    it('should return bad request if body is not valid JSON', function (done){
+      request.post('/note')
+        .send('I am not JSON')
+        .expect(400)
+        .expect('Bad Request : Error: expected item')
```

Comment:
This is not the error message I would expect for invalid JSON. Are you sure the body is being parsed?